assets/tests/hooks/useNotificationsReducer.test.tsx

```diff
@@ -170,4 +192,228 @@ describe("useNotificationsReducer", () => {
 
     expectPut("/api/notification_read_state?new_state=read&notification_ids=1")
   })
+
+  test("opens a channel and processes any initial notifications, unsets the initial render flag via callback", () => {
+    const initialNotifications = [notification1Data]
+    const mockSetIsInitialLoad = jest.fn()
+    const mockSocket = makeMockSocket()
+    const mockChannel = makeMockChannel("ok", {
+      data: {
+        initial_notifications: initialNotifications,
+      },
+    })
+    mockSocket.channel.mockImplementationOnce(() => mockChannel)
+
+    const { result } = renderHook(
+      () => useNotificationsReducer(true, mockSetIsInitialLoad),
+      {
+        wrapper: ({ children }) => (
+          <Wrapper socket={mockSocket} selectedRouteIds={["route"]}>
+            {children}
+          </Wrapper>
+        ),
+      }
+    )
+
+    const [state] = result.current
+
+    expect(mockChannel.join).toHaveBeenCalled()
+    expect(state.notifications).toStrictEqual([notification1])
+    expect(mockSetIsInitialLoad).toHaveBeenCalledWith(false)
+  })
+
+  test("adds new notifications", () => {
+    const mockSetIsInitialLoad = jest.fn()
+    const mockSocket = makeMockSocket()
+    const mockChannel = makeMockChannel()
+    mockSocket.channel.mockImplementationOnce(() => mockChannel)
+
+    mockChannel.on.mockImplementation((_event, dataHandler) => {
+      dataHandler({
+        data: notification1Data,
+      })
+    })
+
+    const { result } = renderHook(
+      () => useNotificationsReducer(true, mockSetIsInitialLoad),
+      {
+        wrapper: ({ children }) => (
+          <Wrapper socket={mockSocket} selectedRouteIds={["route"]}>
+            {children}
+          </Wrapper>
+        ),
+      }
+    )
+
+    const [state] = result.current
+
+    expect(state.notifications).toStrictEqual([notification1])
+    expect(tagManagerEvent).toHaveBeenCalledWith("notification_delivered")
+  })
+
+  test("leaves the channel on unmount", () => {
+    const mockSetIsInitialLoad = jest.fn()
+    const mockSocket = makeMockSocket()
+    const mockChannel = makeMockChannel()
+    mockSocket.channel.mockImplementationOnce(() => mockChannel)
+
+    const { unmount } = renderHook(
+      () => useNotificationsReducer(false, mockSetIsInitialLoad),
+      {
+        wrapper: ({ children }) => (
+          <Wrapper socket={mockSocket} selectedRouteIds={[]}>
+            {children}
+          </Wrapper>
+        ),
+      }
+    )
+
+    expect(mockChannel.join).toHaveBeenCalled()
+
+    unmount()
+
+    expect(mockChannel.leave).toHaveBeenCalled()
+  })
+
+  test("console.error on join error", async () => {
```

Comment:
**question:** this and a few other places seem to effectively duplicate tests from `useChannel`. Is there a reason to be testing these properties here as well?

Reply:
Good point, removed in [c2bd7f2](https://github.com/mbta/skate/pull/1765/commits/c2bd7f2ebc9b4036021c2f05a14b1f2aa72ec8cf)